app/cli/lib/context_load.go

```
@@ -109,176 +137,393 @@ func MustLoadContext(resources []string, params *types.LoadContextParams) {
+				if (params.DefsOnly || params.NamesOnly) && numPaths > shared.MaxContextMapPaths {
+					onErr(fmt.Errorf("too many files to load (found %d, limit is %d)", numPaths, shared.MaxContextMapPaths))
+				} else if !params.DefsOnly && !params.NamesOnly && numPaths > shared.MaxContextCount {
+					onErr(fmt.Errorf("too many files to load (found %d, limit is %d)", numPaths, shared.MaxContextCount))
+				}
 
-		} else {
-			flattenedPaths, err := ParseInputPaths(inputFilePaths, params)
-			if err != nil {
-				onErr(fmt.Errorf("failed to parse input paths: %v", err))
-			}
+				inputFilePaths = flattenedPaths
```

golangci-lint (1.64.8)

ineffectual assignment to inputFilePaths

(ineffassign)